File: 147-insertion-sort-list/insertion-sort-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode insertionSortList(ListNode head) {
        ListNode curr = head;
        ListNode prev = null;
        
        while(curr != null){
            if(prev != null && prev.val > curr.val){
                ListNode temp = head;
                while(temp != curr){
                    if(temp.val > curr.val){
                        int val = temp.val;
                        temp.val = curr.val;
                        curr.val = val;
                    }
                    temp = temp.next;
                }
            }
            prev = curr;
            curr = curr.next;
        }
        return head;
    }
}

File: 147-insertion-sort-list/insertion-sort-list.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var insertionSortList = function(head) {
    let curr = head;
    let prev = null;
    
    while(curr){
        if(prev && prev.val > curr.val){
            let temp = head;
            while(temp != curr){
                if(temp.val > curr.val){
                    let val = temp.val;
                    temp.val = curr.val;
                    curr.val = val;
                }
                temp = temp.next;
            }
        }
        prev = curr;
        curr = curr.next;
    }
    return head;
};